cumulative changes by Julius Mittler

juliusmittler opened the pull request:
All changes done by Julius Mittler since October 2024

includes:
- custom macro creation
- sub workflows from mulit-selection
- macro node expansion
- auto sorting of expanded nodes

juliusmittler committed:
custom_macro creation part 1

added custom_macro creation for a selection of nodes and edges

juliusmittler committed:
custom_macro creation part 2

added custom_macro creation for a selection of nodes and edges

juliusmittler committed:
updated pyironflow to current state

juliusmittler committed:
updated pyironflow to current state

juliusmittler committed:
delete temporary files

juliusmittler committed:
added jupyter notebook "macro"

for development only

juliusmittler committed:
updated gitignore

juliusmittler committed:
ghostNodes and node sizes

added ghostNodes and size dependant layout

juliusmittler committed:
Update before switching Base

juliusmittler committed:
update

update

juliusmittler committed:
update

update

juliusmittler committed:
modified for macros

startet modifying for macro sub nodes

juliusmittler committed:
basic macro nodes

added macro nodes, macro sub nodes
and basic sorting

juliusmittler committed:
update

intermitten update

juliusmittler committed:
in/out demo

demo for in/out nodes for macros

juliusmittler committed:
update

update

juliusmittler committed:
update

made in/output nodes adaptable to macros

juliusmittler committed:
selection/temp_sub_node_list

updated node_selection
added temp_sub_node_list for saving subnode positions

juliusmittler committed:
autosorting macros

added autosorting macros on expansion

juliusmittler committed:
Update for Transfer

juliusmittler committed:
updated to current version

juliusmittler committed:
update to current version

juliusmittler committed:
cleanup

some slight cleanup on the gui and log messages

juliusmittler committed:
Revert "cleanup"

This reverts commit 3dc80183230bebec3a49a2843c9963d7d3913ad1.

juliusmittler committed:
Revert "update to current version"

This reverts commit a7ae4d6f1fb0bd11f1e8bbc9979bf1865b6226a0.

juliusmittler committed:
Revert "updated to current version"

This reverts commit f7c2cc7aaec0ed809c7ddc63a33fe03953bbe0e8.

juliusmittler committed:
cleanup

removed test buttons
arranged sorting functionality
removed some log messages

juliusmittler committed:
cleanup + MacroInOutNode

slight log message cleanup
created basic node type MacroInOutNodes

juliusmittler committed:
updated to main

liamhuber commented:
@juliusmittler, I'll be doing some infrastructure work in this repo this week. I've taken a look at the pending files here, and I think the only place we're likely to wind up with a git conflict is in the gitignore itself; the one I just merged to main is a superset of the one you have here, so you should be able to safely take the one on `main`.

On that note, there's a bunch of files here (e.g. *.pyc and windows files) that we don't want in the repo. When you get the time, would you please remove them? I don't have the operations memorized, but Claude's recommendation looks good to me:

```
git rm --cached $(git ls-files -i --exclude-standard)
git commit -m "Remove files now in .gitignore"
git push
```

Now, that still leaves the files in the git history. Getting rid of that explicitly would require `git filter-branch` and force-pushing. These files aren't a security risk, just data crud, so that seems like more work than it's worth. Instead, we could simply squash-merge this PR when the time comes, and the main history won't ever see these files that were added then removed on this branch alone.

liamhuber commented:
@juliusmittler, I don't know if I'm familiar enough yet with the repo to offer a very in-depth review, but the first thing to do is git-remove all the extra stuff currently in the main branch's [gitignore](https://github.com/pyiron/pyironFlow/blob/main/.gitignore). You might need to force-push to get those files un-tracked here. Next is to merge in main -- probably by uniformly accepting main's `demo.ipynb`; for `.gitignore` you may want to do a real merge and keep some of the stuff you added, idk.

Once the PR is trimmed down to "real" files, please ping me and I'll come give what review I can.

juliusmittler committed:
Loops and cleanup

added simple loop creation/visualisation
removed unwanted files
added multiple node and edge types to support loops

juliusmittler committed:
removed notebooks

removed jupyter notebooks

juliusmittler committed:
trimming files down

removed pyiron_nodes and updated gitignore

juliusmittler committed:
Update .gitignore

juliusmittler committed:
cleanup

juliusmittler committed:
for-loops

implemented for-loop and for loop creation box

juliusmittler committed:
loop update

updated loop creation and view
some bugfixing
